src/psyclone/dynamo0p3.py

```diff
@@ -6162,8 +6137,7 @@ def _lower_bound_fortran(self):
                 raise GenerationError(
                     "Unsupported lower bound name '{0}' "
                     "found".format(self._lower_bound_name))
-            mesh_obj_name = self._name_space_manager.create_name(
-                root_name="mesh", context="PSyVars", label="mesh")
+            mesh_obj_name = self.root.symbol_table.lookup_tag("mesh").name
```

Comment:
Are we guaranteed for this to already exist? If so, why not use name_from_tag?

Reply:
It is probably not guaranteed, so as recommended I changed it to `name_from_tag` for safety.